### PR TITLE
Added the favourite listings UI and the bookmark button to the listin…

### PR DESCRIPTION
**Pull request by akashd50**
- Added the favourite listings UI and the bookmark button to the listing cards and the listing view
- image
- image
- image

### CODE REVIEW

**Comment by akashd50** [comment]
https://github.com/gilhrpenner/COMP4350/issues/113

**Comment by akashd50** [comment]
Do we also wanna have the bookmark icon show up on the tiles on this page? 
image

File: frontend/src/pages/dashboard/FavouriteListings.vue [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            message: i18n.t('couldNotGetFavouriteListings'),
+            icon: INotifyIcon.warn,
+            color: INotifyColor.negative,
+            duration: 10000,
+        });
+
+        return [];
+    });
+
+    console.log('Favourites', favouriteListings.value);
```
Comment (BrettDowneyuofm) [inline frontend/src/pages/dashboard/FavouriteListings.vue:82]: just noting console statement

File: frontend/src/pages/dashboard/FavouriteListings.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
+<template>
+    <q-dialog v-model="showConfirmRemoveDialog" persistent>
```
Comment (BrettDowneyuofm) [inline frontend/src/pages/dashboard/FavouriteListings.vue:2]: showConfirmRemoveDialog does not exist

File: frontend/src/pages/main/View.vue [hunk 3]
```diff
@@ -151,12 +157,57 @@ const getConditionInfo = computed(() => {
...
 
-const toggleBookmark = () => {
-    notify({
-        message: 'This feature is not implemented yet! :D',
-        icon: INotifyIcon.negative,
-        color: INotifyColor.negative,
-    });
+const toggleBookmark = (listingId: string) => {
+    if (isFavouriteListing.value) {
+        api.delete('/account/favourite-listings', {
```
Comment (BrettDowneyuofm) [inline frontend/src/pages/main/View.vue:162]: This is executed when favouriting a listing

File: frontend/src/pages/main/View.vue [hunk 4]
```diff
@@ -151,12 +157,57 @@ const getConditionInfo = computed(() => {
...
+            isFavouriteListing.value = true;
+
+            notify({
+                color: INotifyColor.positive,
+                icon: INotifyIcon.positive,
+                message: i18n.t('listingAddedToFavourites'),
+            });
+
+            const favourites = account.value.favouriteListings;
+            favourites.push(listingId)
```
Comment (BrettDowneyuofm) [inline frontend/src/pages/main/View.vue:201]: Error caused from mutating store state outside of mutation handler: "Error: [vuex] do not mutate vuex store state outside mutation handlers."

**Comment by BrettDowneyuofm** [comment]
Looks good just a few small issues 👍

File: frontend/src/layouts/MainLayout.vue [hunk 5]
```diff
@@ -74,7 +75,15 @@ import { useI18n } from 'vue-i18n';
 const { locale } = useI18n();
 const store = useStore();
 
+const app = computed(() => store.state.app);
+
 onMounted(() => {
+    // Set authorization header whenever user refresh the page
+    if (app.value.isAuthenticated && app.value.authToken) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        api.defaults.headers.common['Authorization'] = `Bearer ${app.value.authToken}`;
```
Comment (BrettDowneyuofm) [inline frontend/src/layouts/MainLayout.vue:84]: I believe this is no longer needed

File: frontend/src/components/BrowseListingItem.vue [hunk 6]
```diff
@@ -7,7 +7,20 @@
...
-                    <div class="listing-card__body--title q-mb-md">{{ listing.title }}</div>
+                    <div class="listing-card__body--title-row q-mb-md">
+                        <div class="listing-card__body--title-row--title">{{ listing.title }}</div>
+                        <q-space />
+                        <q-btn
+                            v-if="showFavouriteIcon"
+                            class="listing-card__body--title-row--bookmark"
+                            flat
+                            round
+                            @click.stop.prevent="$emit('favourite', listing.id, listing.c)"
```
Comment (BrettDowneyuofm) [inline frontend/src/components/BrowseListingItem.vue:18]: listing.c is undefined

**Comment by gilhrpenner** [comment]
> Do we also wanna have the bookmark icon show up on the tiles on this page? image

if that's a quick change, sure, it would look good :D
Would you do that at the top corner of the picture itself or below the image with the info about it?

**Comment by akashd50** [comment]
> 

I was thinking it'll be at the same place, like beside the listing title, since thats where its already available, I'll just need to turn it on, but if it makes more sense to move it to the top, I can look into it!

File: frontend/src/layouts/MainLayout.vue (hunk 5 above)
Comment (akashd50) [reply frontend/src/layouts/MainLayout.vue:84]: is there a particular reason for this, I was getting issues when I went to the main layout and refreshed the page, it wouldn't have the auth token on the requests

File: frontend/src/pages/dashboard/FavouriteListings.vue (hunk 1 above)
Comment (akashd50) [reply frontend/src/pages/dashboard/FavouriteListings.vue]: Fixed in 1460ba2ae4ae9b470bf4020aef0999c09bab709b!

File: frontend/src/pages/dashboard/FavouriteListings.vue (hunk 2 above)
Comment (akashd50) [reply frontend/src/pages/dashboard/FavouriteListings.vue]: Fixed in 1460ba2ae4ae9b470bf4020aef0999c09bab709b!

File: frontend/src/components/BrowseListingItem.vue (hunk 6 above)
Comment (akashd50) [reply frontend/src/components/BrowseListingItem.vue]: Fixed in 1460ba2ae4ae9b470bf4020aef0999c09bab709b!

**Review by gilhrpenner — APPROVED** [review]

File: frontend/src/layouts/MainLayout.vue (hunk 5 above)
Comment (gilhrpenner) [reply frontend/src/layouts/MainLayout.vue:84]: If you have recently rebased this branch onto master you shouldn't have issues. One of @BrettDowneyuofm latest PRs changed this and moved it to a different file.

**Comment by gilhrpenner** [comment]
I gave my approval but would rather Brett or Mark to also review this since I've been off from the previous PRs.

File: frontend/src/layouts/MainLayout.vue (hunk 5 above)
Comment (akashd50) [reply frontend/src/layouts/MainLayout.vue]: Alrighty, I've removed it after rebasing master!

**Review by BrettDowneyuofm — APPROVED** [review]

**Pull request merged**